qemu/tests/x86_cpu_model_boot.py

```diff
@@ -0,0 +1,149 @@
+    for flag in flags_dict.keys():
+        tmp_list = []
+        tmp_list.extend(list(map(lambda x: flag.replace(*x), replace_char)))
+        for tmp in tuple(tmp_list):
+            if tmp != flag:
+                tmp_dict[tmp] = flags_dict[flag]
+    return dict(flags_dict, **tmp_dict)
+
+
+def get_cpu_info_from_dumpxml(name):
+    """
+    Get cpu info from virsh dumpxml
+    :param name: Domain name
+    :return: Cpu info dict in dumpxml
+    """
+    cpu_xml = libvirt_xml.VMXML.new_from_dumpxml(name).cpu
+    feature_list = cpu_xml.get_feature_list()
+    cpu_model = cpu_xml.model
+    cpu_features = {}
+    for i in range(0, len(feature_list)):
+        feature_name = cpu_xml.get_feature(i).get('name')
+        feature_policy = cpu_xml.get_feature(i).get('policy')
+        if feature_policy == "require":
+            feature_policy = "on"
+        elif feature_policy == "disable":
+            feature_policy = "off"
+        cpu_features[feature_name] = feature_policy
+    cpu_info = {}
+    cpu_info["model"] = cpu_model
+    cpu_info["features"] = cpu_features
+    return cpu_info
+
+
+def compare_cpu_flags(test, params):
+    """
+    Compare flags between qemu cli and libvirt dumpxml with 'host-model'
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :return: True or False
+    """
+    vm_arch = params["vm_arch_name"]
+    if vm_arch not in ['x86_64', 'i686']:
+        raise NotImplementedError("Arch '%s' is not supported" % vm_arch)
+    machine = params.get('machine_type')
+    name = uuid.uuid4().hex
+    try:
+        path.find_command("virsh")
+    except path.CmdNotFoundError:
+        test.cancel("Virsh executable not set or found on path")
+
+    xml = """
+            <domain type='kvm'>
+                <name>%s</name>
+                <memory>1</memory>
+                <os>
+                    <type arch='%s' machine='%s'>hvm</type>
+                </os>
+                <cpu mode='host-model'/>
+            </domain>
+          """ % (name, vm_arch, machine)
+    xml_file = os.path.join(data_dir.get_tmp_dir(), "temp_xml_for_cpu")
+    with open(xml_file, "w") as f:
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        logging.error(err)
+        return
+    else:
+        qemu_cpu_info = cpu.parse_qemu_cpu_flags(
+            cpu.get_cpu_info_from_virsh_qemu_cli(name)['flags'])
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)['features']
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        qemu_cpu_info = get_flags_patterns(qemu_cpu_info)
+        for flag in libvirt_cpu_info.keys():
+            if flag not in qemu_cpu_info.keys():
+                if libvirt_cpu_info[flag] == 'on':
+                    logging.error("miss some flags from qemu cli")
```

Comment:
Prefer to collect all the missing or mismatch flags and print in the log for better debug.

Reply:
> Prefer to collect all the missing or mismatch flags and print in the log for better debug.

update and thanks.